src/index.ts

```diff
@@ -8,37 +8,46 @@ const cssResolutionUnits: string[] = require('css-resolution-units');
 const cssFrequencyUnits: string[] = require('css-frequency-units');
 const cssTimeUnits: string[] = require('css-time-units');
 
+const numberPrefixPattern = /^(\+|-)?(\.)?\d/;
```

Comment:
Why is this so far away from the implementation? Also, I can remove the trailing `\d` and all the tests still pass.

Reply:
yes, there needs to be be test added for this:
without this you could enter "NaN%" and it would falsely validate it a  number